cmd/agent/api/server.go

```diff
@@ -125,10 +130,6 @@ func StartServer() error {
 		ErrorLog: stdLog.New(&config.ErrorLogWriter{
 			AdditionalDepth: 5, // Use a stack depth of 5 on top of the default one to get a relevant filename in the stdlib
 		}, "Error from the agent http API server: ", 0), // log errors to seelog,
-		// TODO: WriteTimeout closes gRPC streams every $server_timeout
-		// seconds, need to find a solution for that before
-		// re-enabling.
-		// WriteTimeout: config.Datadog.GetDuration("server_timeout") * time.Second,
```

Comment:
I think this line `WriteTimeout: config.Datadog.GetDuration("server_timeout") * time.Second,` should be re-added instead of removed

Reply:
I would agree, and that indeed was my first try, but it didn't work :) See https://github.com/golang/go/issues/16100 for (a whole lot of) context. In short, `WriteTimeout` is for the HTTP connection, and `SetWriteDeadline` changes the deadline on the underlying TCP connection, making the change basically ineffective.

Reply:
Interesting. From poking around in the `server.go` source it looks like `WriteTimeout` is only ever used to call `SetWriteDeadline`.
https://github.com/golang/go/blob/master/src/net/http/server.go#L974
https://github.com/golang/go/blob/master/src/net/http/server.go#L1837

So I would expect the behavior of uncommenting this to be the same as setting the deadline manually (unless I am missing something where the HTTP connection is handled differently). 

FWIW this didn't seem to be an issue in #7163 - where I initiate an HTTP connection, then override the timeout and hold the socket open till canceled (however I am not familiar enough with `grpcHandlerFunc` to argue that it should work the same way). 

If the behavior going forward is to not have a timeout at all, and then selectively add the timeout to calls that need it - we should probably audit existing calls and figure out which ones require a timeout (since previously - it was all of them).

Reply:
Note that ConnContext does not operate in the HTTP connection, but the underlying TCP one here: https://github.com/golang/go/blob/master/src/net/http/server.go#L3004 -- so we're not setting timeouts on the same connection.

In any case, this does not remove timeouts entirely, only for gRPC calls. `grpcHandlerFunc` adds the timeouts for all HTTP endpoints, while still allowing opt-outs, so this is very close to the behavior before I commented out the `WriteTimeout`, the only difference being that gRPC now has no timeouts at all.

Reply:
got it. Sounds good - thanks for clarifying :)